rpc/filter_api.go

```diff
@@ -0,0 +1,208 @@
+package rpc
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/ethermint/x/evm/types"
+
+	"math/big"
+
+	eth "github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+type FilterCriteria eth.FilterQuery
```

Comment:
What is the reason that Ethereum's type could not be used as is? FilterCriteria type is not in internal package and seems unmodified

Reply:
You can only define methods on a type defined in that same package. So the `UnmarshalJSON` receiver function that requires `eth.FilterQuery` can be used appropriately within the same package.

Reply:
I'm saying to just use Ethereum's type. It's only used for unmarshalling to be used and you don't introduce any new functionality in this PR. My suggestion I just did in a commit here: https://github.com/ChainSafe/ethermint/commit/a2dfd0bfafed74f70d0162ab6fe10061809aa281 let me know why this doesn't work for what you're doing

Reply:
I see! Yeah, miscommunication there. For sure will make the change.